pkg/kvstore/consul.go

```diff
@@ -389,12 +389,14 @@ func (c *ConsulClient) GetWatcher(key string, timeSleep time.Duration) <-chan []
 			}
 			curSeconds = time.Second
 			qo.WaitIndex = q.LastIndex
-			go func() {
+			cpyValue := make([]byte, len(k.Value))
+			copy(cpyValue, k.Value)
+			go func(cpyValue []byte) {
```

Comment:
I assume the purpose of the go subroutine is to avoid blocking on the channel. Does it make sense to instead use a larger buffer for the channel? You are essentially building a buffer using go subroutines and have no limit whatsoever.

At the very least, this needs a good comment explaining why this is done one way or another.